common/transport/http/src/http.ts

```diff
@@ -144,6 +149,14 @@ export class Http {
     return msg;
   }
 
+  /**
+   * @private
+   */
+  setOptions(options: any, callback: (err?: Error) => void): void {
```

Comment:
>setOptions(options: any, callback: (err?: Error) => void): void { [](start = 2, length = 65)

why make this asynchronous? no need now, and since it's private we can always change it later.